sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/JoinOptimizationSuite.scala

```diff
@@ -59,12 +75,7 @@ class JoinOptimizationSuite extends PlanTest {
           (noCartesian, seq_pair._2)
         }
       }
-      testExtractCheckCross(plan, expectedNoCross)
-    }
-
-    def testExtractCheckCross
-        (plan: LogicalPlan, expected: Option[(Seq[(LogicalPlan, InnerLike)], Seq[Expression])]) {
-      assert(ExtractFiltersAndInnerJoins.unapply(plan) === expected)
```

Comment:
since you rewrite this function, is the previous comparison logic wrong?

Reply:
If we have multiple conditions, we need the compare them as `Set`.